x/perpetual/keeper/begin_blocker.go

```diff
@@ -69,4 +69,37 @@ func (k Keeper) BeginBlocker(ctx sdk.Context) {
 		}
 		k.SetPool(ctx, pool)
 	}
+
+	// Handle toPay
+	err := k.HandleToPay(ctx)
+	if err != nil {
+		ctx.Logger().Error(err.Error())
+	}
+}
+
+func (k Keeper) HandleToPay(ctx sdk.Context) error {
```

Comment:
@amityadav0 can we move that logic from ABCI to the "closed position" offchain agent flow? that way we would reduce the per block computation complexity

Reply:
yes, moved